permission/permission.go

```diff
@@ -0,0 +1,759 @@
+		if _, err := os.OpenFile(path, os.O_CREATE|os.O_RDWR, 0644); err != nil {
+			log.Error("Failed to create disallowed-nodes.json file", "err", err)
+			return
+		}
+		fileExisted = false
+	}
+
+	nodelist := []string{}
+	// Load the nodes from the config file
+	if fileExisted == true {
+		blob, err := ioutil.ReadFile(path)
+		if err != nil {
+			log.Error("updateDisallowedNodes Failed to access disallowed-nodes.json", "err", err)
+			return
+		}
+		if blob != nil {
+			if err := json.Unmarshal(blob, &nodelist); err != nil {
+				log.Error("updateDisallowedNodes: Failed to load nodes list", "err", err)
+				return
+			}
+		}
+	}
+
+	nodelist = append(nodelist, url)
+	blob, _ := json.Marshal(nodelist)
+
+	p.mux.Lock()
+	defer p.mux.Unlock()
+
+	if err := ioutil.WriteFile(path, blob, 0644); err != nil {
+		log.Error("updateDisallowedNodes: Error writing new node info to file", "err", err)
+	}
+	// Disconnect the peer if it is already connected
+	p.disconnectNode(url)
+}
+
+// Monitors account access related events and updates the cache accordingly
+func (p *PermissionCtrl) manageAccountPermissions() {
+	chAccessModified := make(chan *pbind.AcctManagerAccountAccessModified)
+	chAccessRevoked := make(chan *pbind.AcctManagerAccountAccessRevoked)
+	chStatusChanged := make(chan *pbind.AcctManagerAccountStatusChanged)
+
+	var evtAccessModified *pbind.AcctManagerAccountAccessModified
+	var evtAccessRevoked *pbind.AcctManagerAccountAccessRevoked
+	var evtStatusChanged *pbind.AcctManagerAccountStatusChanged
+
+	opts := &bind.WatchOpts{}
+	var blockNumber uint64 = 1
+	opts.Start = &blockNumber
+
+	if _, err := p.permAcct.AcctManagerFilterer.WatchAccountAccessModified(opts, chAccessModified); err != nil {
+		log.Info("Failed NewNodeProposed", "error", err)
+	}
+
+	if _, err := p.permAcct.AcctManagerFilterer.WatchAccountAccessRevoked(opts, chAccessRevoked); err != nil {
+		log.Info("Failed NewNodeProposed", "error", err)
+	}
+
+	if _, err := p.permAcct.AcctManagerFilterer.WatchAccountStatusChanged(opts, chStatusChanged); err != nil {
+		log.Info("Failed NewNodeProposed", "error", err)
+	}
+
+	for {
+		select {
+		case evtAccessModified = <-chAccessModified:
+			types.AcctInfoMap.UpsertAccount(evtAccessModified.OrgId, evtAccessModified.RoleId, evtAccessModified.Address, evtAccessModified.OrgAdmin, types.AcctStatus(int(evtAccessModified.Status.Uint64())))
+
+		case evtAccessRevoked = <-chAccessRevoked:
+			types.AcctInfoMap.UpsertAccount(evtAccessRevoked.OrgId, evtAccessRevoked.RoleId, evtAccessRevoked.Address, evtAccessRevoked.OrgAdmin, types.AcctActive)
+
+		case evtStatusChanged = <-chStatusChanged:
+			ac := types.AcctInfoMap.GetAccount(evtStatusChanged.Address)
+			types.AcctInfoMap.UpsertAccount(evtStatusChanged.OrgId, ac.RoleId, evtStatusChanged.Address, ac.IsOrgAdmin, types.AcctStatus(int(evtStatusChanged.Status.Uint64())))
+		}
+	}
+}
+
+// Disconnect the node from the network
+func (p *PermissionCtrl) disconnectNode(enodeId string) {
```

Comment:
error is swallowed
better to capture and return an error when disconnecting node

Reply:
have logged the error. This is running in a separate go routing and hence preferred logging of the error.